src/Sarif.Multitool/Rules/SARIF2004.OptimizeFileSize.cs

```diff
@@ -55,6 +71,66 @@ protected override void Analyze(Run run, string runPointer)
         {
             AnalyzeLocationOnlyArtifacts(run, runPointer);
             AnalyzeIdOnlyRules(run, runPointer);
+
+            if (run.Results != null)
+            {
+                AnalyzeResults(run.Results, runPointer.AtProperty(SarifPropertyName.Results));
```

Comment:
>AnalyzeResults [](start = 16, length = 14)

You don't need to write `AnalyzeResults` and call it from here. Just override `Analyze(Result result, string resultPointer)`.

This works because you really want to analyze every result. This is unlike the cases where you only want to analyze _certain_ physical locations, or _certain_ URIs -- and so you have to navigate down from the parent to make sure that you only analyze the ones you care about.

Make sense?

Reply:
Yes!

---
In reply to: [455195243](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455195243) [](ancestors = 455195243)